chia/wallet/trade_manager.py

```diff
@@ -517,15 +524,26 @@ async def _create_offer_for_ids(
                     else:
                         asset_id = id
                         wallet = await self.wallet_state_manager.get_wallet_for_asset_id(asset_id.hex())
+                    assert wallet is not None
                     if not callable(getattr(wallet, "get_coins_to_offer", None)):  # ATTENTION: new wallets
                         raise ValueError(f"Cannot offer coins from wallet id {wallet.id()}")
                     # For the XCH wallet also include the fee amount to the coins we use to pay this offer
                     amount_to_select = abs(amount)
                     if wallet.type() == WalletType.STANDARD_WALLET:
                         amount_to_select += fee
-                    coins_to_offer[id] = await wallet.get_coins_to_offer(
-                        asset_id, uint64(amount_to_select), tx_config.coin_selection_config
-                    )
+                    assert isinstance(wallet, (CATWallet, DataLayerWallet, NFTWallet, Wallet))
+                    if isinstance(wallet, DataLayerWallet):
```

Comment:
I try not to switch on specific wallet types in the trade manager whenever possible.  Probably could have done a better job hinting the `get_coins_to_offer` function in the protocol but I don't see why this change is necessary.